src/components/DateInput/index.js

```diff
@@ -0,0 +1,156 @@
+import React from 'react'
+import styled, { css } from 'styled-components'
+import { space, device } from '../../theme'
+import { Input } from '../Input'
+import { Select } from '../Select'
+import { useDeviceInfo } from '../../hooks'
+
+const Container = styled.div`
+  input[type='date'] {
+    ${props =>
+      props.input !== 'native' &&
+      css`
+        display: none;
+      `};
+  }
+
+  .custom {
+    ${props =>
+      props.input !== 'custom' &&
+      css`
+        display: none;
+      `};
+  }
+`
+
+const InputGroup = styled.div`
+  position: relative;
+  display: flex;
+  grid-auto-flow: row;
+
+  @media ${device.mobile} {
+    grid-auto-flow: column;
+  }
+
+  @supports (display: grid) {
+    display: grid;
+    grid-gap: ${space[8]};
+    justify-items: center;
+    margin: auto;
+  }
+`
+
+const InputWrapper = styled.div`
+  width: 100%;
+`
+
+const SelectWrapper = styled.div`
+  width: 100%;
+  padding: 0 ${space[8]};
+
+  @supports (display: grid) {
+    padding: 0;
+  }
+`
+
+function pad(str) {
+  if (str.length === 1) {
+    return (str = '0' + str)
+  }
+  return str
+}
+
+function createDateString(date) {
+  if (!date) return false
+  let d = date
+  if (typeof date.getMonth !== 'function') {
+    d = new Date(date)
+  }
+  let month = '' + (d.getMonth() + 1)
+  let day = '' + d.getDate()
+  const year = d.getFullYear()
+  month = pad(month)
```

Comment:
You can use the native JS `month.padStart(2, '0')` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/padStart)

Reply:
Haha I looked at String methods, but that one’s not supported in IE :P